### ISSUE TITLE
Idea: @db.register_function(deterministic=True)

### ISSUE THREAD

**Issue by simonw** [simonw/sqlite-utils#191]
Python 3.8 added a `deterministic` parameter to `db.create_function()`: https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_function

`sqlite-utils` could expose this in the decorator, only actually applying it if the Python version supports it (using feature detection) - since nothing will break if it's not applied.

https://sqlite-utils.readthedocs.io/en/stable/python-api.html#registering-custom-sql-functions

```python
db = Database(memory=True)

@db.register_function(deterministic=True)
def reverse_string(s):
    return "".join(reversed(list(s)))
```

**Comment by simonw** [comment]
https://stackoverflow.com/a/3931903 looks useful:
```python
def trace(*args):
    def _trace(func):
        def wrapper(*args, **kwargs):
            print enter_string
            func(*args, **kwargs)
            print exit_string
        return wrapper
    if len(args) == 1 and callable(args[0]):
        # No arguments, this is the decorator
        # Set default values for the arguments
        enter_string = 'entering'
        exit_string = 'exiting'
        return _trace(args[0])
    else:
        # This is just returning the decorator
        enter_string, exit_string = args
        return _trace
```
Can improve that code with `functools.wraps`.

**Comment by simonw** [comment]
I think I made a mistake when I designed the initial decorator. I should have had it work like this:

```python
@db.register_function()
def reverse_string(s):
    return "".join(reversed(list(s)))
```
As this leaves open the option to add new parameters in the future.

To avoid breaking backwards compatibility I'll use the hack that detects the argument this time, but in the future I'll try to remember to always design decorators to be called like `@decorator()`.
